### PR TITLE
Solution to Vaults assignment

### PR DESCRIPTION
**Pull request by ryanlabouve**
* Vault1 works
* Vault2 _works_ but needs more tests and to handle edge cases better

### CODE REVIEW

File: test/Vault2Test.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+			await vault2.connect(addr1).mint({
+				value: ethers.utils.parseEther("1.0")
+			});
+			let addr1Address: string = await addr1.getAddress();
+			let newBalance = await vault2.balanceOf(addr1Address);
+			expect(newBalance).to.eq(ethers.utils.parseEther("1.0"));
+		});
+	});
+
+	describe("#burn", async () => {
```
Comment (shubham-kanodia) [inline test/Vault2Test.ts:35]: You can also check for failure cases
